scripts/build/watch.js

```diff
@@ -0,0 +1,92 @@
+const Gaze = require('gaze').Gaze;
+const { spawn } = require('child_process');
+const rootdir = __dirname + '/../../';
+
+const watchers = [];
+
+const GAZE_OPTIONS = {
+  interval: 1000 // how often the target should be polled in milliseconds
+};
+
+const configs = [
+  // admin css
+  { 
+    files: [ 'admin/src/css/**/*' ],
+    task: 'exec:less',
+  },
+
+  // admin-js
+  {
+    files: [ 'admin/src/js/**/*', 'shared-libs/*/src/**/*' ],
+    task: 'exec:browserify-admin',
+  },
+
+  // admin-templates
+  {
+    files: [ 'admin/src/templates/**/*' ],
+    task: 'exec:compile-admin-templates',
+  },
+
+  // webapp-js
+  // instead of watching the source files which are watched separately, watch the build folder and upload on rebuild
+  {
+    files: [ 'api/build/static/webapp/**/*', '!api/build/static/webapp/service-worker.js' ],
+    task: 'update-service-worker',
+  },
+
+  // api-public-files
+  {
+    files: [ 'api/src/public/**/*' ],
+    task: 'exec:copy-api-resources',
+  },
+
+  // ddocs
+  {
+    files: [ 'ddocs/*-db/**/*' ],
```

Comment:
I'm changing a line in one secondary ddoc and I'm getting this error:
```
/home/diana/projects/medic-second/ddocs/logs-db/logs/views/connected_users/map.js updated...
Running "exec:copy-ddocs" (exec) task
Running "set-ddocs-version" task
Running "set-build-info" task
Warning: ENOTDIR: not a directory, scandir '/home/diana/projects/medic-second/build/ddocs/logs.json' Use --force to continue.
Aborted due to warnings.
```

Reply:
Ok... the problem here was a slight difference in the grunt tasks. Previously the clean build called `set-build-info` but the watch build did not. This was the only difference between the two and I must have for simplicity to make both the same. Unfortunately if the build dir isn't clean then the `json` ddocs are sitting there and our code was trying to readDir which failed.

I couldn't see a clean way to have a single task handle both cases, and still don't like the idea of having two nearly identical tasks, so I've modified the `set-build-info` to ignore files.

Let me know what you think!

Reply:
Cool, I see you added a check before doing scan, this works well. I agree with not wanting two nearly identical tasks.